app/controllers/report_controller.rb

```diff
@@ -12,6 +12,7 @@ class ReportController < ApplicationController
 
   before_action :check_privileges
   before_action :get_session_data
+  before_action :set_flash_messages, :only => :tree_select, :if => proc { @sb.try(:[], :flash_msg) }
```

Comment:
why do you do this through `before_action` and not right in the `tree_select` method?